### DISCUSSION TITLE
Typescript issue from Menu example

### DISCUSSION THREAD

**Discussion by grelas** [adobe/react-spectrum#3955] [General]
Copied and pasted the `Button` code from https://codesandbox.io/s/unruffled-pascal-xo2ilk for `useMenu` and got:

```
Argument of type '(props: ButtonProps, ref: React.RefObject<HTMLButtonElement>) => JSX.Element' is not assignable to parameter of type 'ForwardRefRenderFunction<HTMLButtonElement, ButtonProps>'.
  Types of parameters 'ref' and 'ref' are incompatible.
    Type 'ForwardedRef<HTMLButtonElement>' is not assignable to type 'RefObject<HTMLButtonElement>'.
      Type 'null' is not assignable to type 'RefObject<HTMLButtonElement>'
```

typescript version: 4.1.3

**Comment by snowystinger** [comment]
Looks like that codesandbox is also at TS 4.1.3, and those are TS checked files, so I'm not sure why you're getting an error when there is no error in the codesandbox. Can you provide a reproduction?

**Comment by binaryartifex** [comment]
whenever i use forwardRef with anything react-aria, the first line in the component is always (using button component in this instance) is

```typescript
import { useObjectRef } from "@react-aria/utils";

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
    let buttonRef = useObjectRef(ref);
    let { buttonProps, isPressed } = useButton(props, buttonRef);
    let { focusProps, isFocusVisible } = useFocusRing();
    
    // ...rest of component
```

i havnt had any typescript ref issues since.

**Reply by snowystinger** [reply]
Ah, yes, right now this is expected. See https://github.com/adobe/react-spectrum/issues/834 for more details
